packages/plugin/src/Integrations/PaymentGateways/PayPal/EventListeners/RenderTemplates.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\EventListeners;
+
+use Solspace\Freeform\Elements\Submission;
+use Solspace\Freeform\Events\Submissions\RenderSubmissionFieldEvent;
+use Solspace\Freeform\Events\Submissions\RenderTableValueEvent;
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\Fields\PayPalField;
+use Solspace\Freeform\Library\Bundles\FeatureBundle;
+use Solspace\Freeform\Records\Pro\Payments\PaymentRecord;
+use Solspace\Freeform\Services\SubmissionsService;
+use yii\base\Event;
+
+class RenderTemplates extends FeatureBundle
+{
+    public function __construct()
+    {
+        Event::on(
+            SubmissionsService::class,
+            SubmissionsService::EVENT_RENDER_FIELD,
+            [$this, 'renderSubmissionFieldTemplate']
+        );
+
+        Event::on(
+            Submission::class,
+            Submission::EVENT_RENDER_TABLE_VALUE,
+            [$this, 'renderTableValue']
+        );
+    }
+
+    public function renderSubmissionFieldTemplate(RenderSubmissionFieldEvent $event): void
+    {
+        $field = $event->getField();
+        if (!$field instanceof PayPalField) {
+            return;
+        }
+
+        $payment = PaymentRecord::findOne([
+            'submissionId' => $event->getSubmission()->id,
+            'fieldId' => $field->getId(),
+        ]);
+
+        if (!$payment) {
+            $event->setOutput('-');
+
+            return;
+        }
+
+        $template = file_get_contents(__DIR__.'/../Templates/submission-field.twig');
+        $metadata = json_decode($payment->metadata ?? '{}', true);
+
+        // Prefer metadata for amount/currency/status; fallback to record
+        $unit = $metadata['unit'] ?? 'dollars';
+        $rawAmount = isset($metadata['amount']) ? (float) $metadata['amount'] : (float) ($payment->amount ?? 0);
```

Comment:
This part is pretty unreadable. In this case an `if/else` makes it more readable.

```suggestion
        $rawAmount = 0;
        if (isset($metadata['amount'])) {
            $rawAmount = (float) $metadata['amount'];
        } else if (isset($payment->amount)) {
            $rawAmount = $payment->amount;
        }
```